concrete/src/Block/Block.php

```diff
@@ -1596,23 +1597,36 @@ public function move(Collection $collection, Area $area)
      */
     public function duplicate($nc, $isCopyFromMasterCollectionPropagation = false)
     {
-        $db = Loader::db();
-        $dh = Loader::helper('date');
+        $app = Facade::getFacadeApplication();
+        /** @var Connection $connection */
+        $connection = $app->make(Connection::class);
+        /** @var Date $dh */
+        $dh = $app->make('helper/date');
 
         $bt = BlockType::getByID($this->getBlockTypeID());
         $blockTypeClass = $bt->getBlockTypeClass();
         if (!$blockTypeClass) {
             return false;
         }
-        $app = Facade::getFacadeApplication();
         $bc = $app->make($blockTypeClass, ['obj' => $this]);
 
         $bDate = $dh->getOverridableNow();
-        $v = [$this->getBlockName(), $bDate, $bDate, $this->getBlockFilename(), $this->getBlockTypeID(), $this->getBlockUserID()];
-        $q = 'insert into Blocks (bName, bDateAdded, bDateModified, bFilename, btID, uID) values (?, ?, ?, ?, ?, ?)';
-        $r = $db->prepare($q);
-        $res = $db->execute($r, $v);
-        $newBID = $db->Insert_ID(); // this is the latest inserted block ID
+        $connection->createQueryBuilder()
```

Comment:
I love the `insert` method much more ;)

It'd be
```php
$connection->insert('Blocks', [
    'bName' => $this->getBlockName(),
    'bDateAdded' => $bDate,
    'bDateModified' => $bDate,
    'bFilename' => $this->getBlockFilename(),
    'btID' => $this->getBlockTypeID(),
    'uID', $this->getBlockUserID(),
]);
```

which is much more readable

Reply:
Yeah, I agree, I find these query. builder statements extremely difficult to read.

Reply:
@mlocati @aembler Changed to use `$connection->insert()`